ctapipe/image/image_processor.py

```diff
@@ -209,51 +211,48 @@ def _parameterize_image(
         # parameterization
         return default
 
-    def _process_telescope_event(self, event):
+    def _process_telescope_event(self, tel_event: TelescopeEventContainer):
         """
         Loop over telescopes and process the calibrated images into parameters
         """
-        for tel_id, dl1_camera in event.dl1.tel.items():
+        tel_id = tel_event.index.tel_id
+        dl1 = tel_event.dl1
 
-            if self.apply_image_modifier.tel[tel_id]:
-                dl1_camera.image = self.modify(tel_id=tel_id, image=dl1_camera.image)
+        if self.apply_image_modifier.tel[tel_id]:
+            dl1.image = self.modify(tel_id=tel_id, image=dl1.image)
 
-            dl1_camera.image_mask = self.clean(
-                tel_id=tel_id,
-                image=dl1_camera.image,
-                arrival_times=dl1_camera.peak_time,
-            )
+        dl1.image_mask = self.clean(
+            tel_id=tel_id,
+            image=dl1.image,
+            arrival_times=dl1.peak_time,
+        )
+        dl1.parameters = self._parameterize_image(
+            tel_id=tel_id,
+            image=dl1.image,
+            signal_pixels=dl1.image_mask,
+            peak_time=dl1.peak_time,
+            default=self.default_image_container,
+        )
 
-            dl1_camera.parameters = self._parameterize_image(
-                tel_id=tel_id,
-                image=dl1_camera.image,
-                signal_pixels=dl1_camera.image_mask,
-                peak_time=dl1_camera.peak_time,
-                default=self.default_image_container,
+        self.log.debug("params: %s", dl1.parameters.as_dict(recursive=True))
+
+        if (
```

Comment:
Why do we test for tel_event.simulation here? I struggle to understand the practical case. Would it be valid if true_image is not None but tel_event.simulation is None?

Reply:
There are observed events, where `event.simulation is None` and there are simulations, which don't have the `true_image` information, so `event.simulation` is there, but `event.simulation.true_image` is None.

We need to test both.

Reply:
And python only evaluates the second expression if the first ist true, this is why we need to check first `tel_event.simulation is not None` and then (which will only be checked if we do have simulation) check if the true image is available